### PR TITLE
Updating the documentation for System.Runtime

### PR DESCRIPTION
**Pull request by tannergooding**
This helps resolve most of https://github.com/dotnet/runtime/issues/105987

### CODE REVIEW

File: xml/System/MemoryExtensions.xml [hunk 1]
```diff
@@ -9552,12 +9556,17 @@ Invalid sequences will be represented in the enumeration by <xref:System.Text.Ru
...
-        <param name="source">To be added.</param>
-        <param name="separators">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="T">The type of the elements.</typeparam>
+        <param name="source">The source span to be enumerated.</param>
+        <param name="separators">The <see cref="T:System.MemoryExtensions.SpanSplitEnumerator`1" /> to be used to split the provided span.</param>
+        <summary>
+          <para>Returns a type that allows for enumeration of each element within a split span using the provided <see cref="T:System.MemoryExtensions.SpanSplitEnumerator`1" />.</para>
```
Comment (gewarren) [inline xml/System/MemoryExtensions.xml:9563]:
```suggestion
          <para>Returns a type that allows for enumeration of each element within a split span using the provided separator characters.</para>
```

File: xml/System/MemoryExtensions.xml [hunk 2]
```diff
@@ -9552,12 +9556,17 @@ Invalid sequences will be represented in the enumeration by <xref:System.Text.Ru
...
+        </summary>
+        <returns>Returns a <see cref="T:System.MemoryExtensions.SpanSplitEnumerator`1" />.</returns>
```
Comment (gewarren) [inline xml/System/MemoryExtensions.xml:9565]:
```suggestion
        <returns>An enumerator that can be used to enumerate each element in a split span using the provided separator characters.</returns>
```

File: xml/System/MemoryExtensions.xml [hunk 3]
```diff
@@ -9552,12 +9556,17 @@ Invalid sequences will be represented in the enumeration by <xref:System.Text.Ru
...
+        <remarks>
+          <para>Unlike <see cref="M:System.MemoryExtensions.SplitAny``1(System.ReadOnlySpan{``0},System.ReadOnlySpan{``0})" />, the <paramref name="separators" /> is not checked for being empty.</para>
```
Comment (gewarren) [inline xml/System/MemoryExtensions.xml:9567]:
```suggestion
          <para>Unlike <see cref="M:System.MemoryExtensions.SplitAny``1(System.ReadOnlySpan{``0},System.ReadOnlySpan{``0})" />, <paramref name="separators" /> is not checked for being empty.</para>
```

File: xml/System/MemoryExtensions.xml [hunk 4]
```diff
@@ -9552,12 +9556,17 @@ Invalid sequences will be represented in the enumeration by <xref:System.Text.Ru
...
+          <para>An empty <paramref name="separators" /> will result in no separators being found, regardless of the type of <typeparamref name="T" />, whereas <see cref="M:System.MemoryExtensions.SplitAny``1(System.ReadOnlySpan{``0},System.ReadOnlySpan{``0})" /> will use all Unicode whitespace characters as separators if <paramref name="separators" /> is empty and <typeparamref name="T" /> is <see cref="T:System.Char" />.</para>
```
Comment (gewarren) [inline xml/System/MemoryExtensions.xml:9568]:
```suggestion
          <para>An empty <paramref name="separators" /> argument will result in no separators being found, regardless of the type of <typeparamref name="T" />, whereas <see cref="M:System.MemoryExtensions.SplitAny``1(System.ReadOnlySpan{``0},System.ReadOnlySpan{``0})" /> uses all Unicode whitespace characters as separators if <paramref name="separators" /> is empty and <typeparamref name="T" /> is <see cref="T:System.Char" />.</para>
```

File: xml/System/MemoryExtensions.xml [hunk 5]
```diff
@@ -9618,12 +9627,16 @@ Invalid sequences will be represented in the enumeration by <xref:System.Text.Ru
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="T">The type of the elements.</typeparam>
+        <param name="source">The source span to be enumerated.</param>
+        <param name="separators">The separators to be used to split the provided span.</param>
+        <summary>
+          <para>Returns a type that allows for enumeration of each element within a split span using any of the provided elements.</para>
+        </summary>
+        <returns>Returns a <see cref="T:System.MemoryExtensions.SpanSplitEnumerator`1" />.</returns>
```
Comment (gewarren) [inline xml/System/MemoryExtensions.xml:9636]:
```suggestion
        <returns>An enumerator that can be used to enumerate each element in a split span using the provided separator characters.</returns>
```

**Review by gewarren — APPROVED** [review]
I left some suggestions to consider.

File: xml/System/MemoryExtensions.xml [hunk 6]
```diff
@@ -9552,12 +9556,17 @@ Invalid sequences will be represented in the enumeration by <xref:System.Text.Ru
...
       <Docs>
-        <typeparam name="T">To be added.</typeparam>
-        <param name="source">To be added.</param>
-        <param name="separators">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="T">The type of the elements.</typeparam>
+        <param name="source">The source span to be enumerated.</param>
+        <param name="separators">The <see cref="T:System.MemoryExtensions.SpanSplitEnumerator`1" /> to be used to split the provided span.</param>
```
[the remark this answers is not in the document]
Comment (tannergooding) [reply xml/System/MemoryExtensions.xml:9561]:
```suggestion
        <param name="separators">The <see cref="T:System.Buffers.SearchValues`1" /> to be used to split the provided span.</param>
```

**Review by krwq — APPROVED** [review]

**Pull request merged**